Add Markdownlint

Gijsreyn opened the pull request:
This pull request adds the markdownlint settings, including a workflow file. And yes, Mikey, happy to see you blogging again.

Gijsreyn committed:
(GH-4) Add linting for Markdown syntax

Prior to this change, the repository had no static analysis for
Markdown syntax. This change:

- Resolves #4
- Defines configuration for markdownlint:
  - `.markdownlint.yaml` contains configuration for the rules, defining
    how we lint the syntax itself.
  - `.markdownlint-cli2.yaml` contains configuration for the CLI, which
    informs CI and how we test locally.
- Adds a linting GHA workflow with a job for linting Markdown syntax
  with markdownlint. This action only runs against changed Markdown
  files.
- Adds a markdown snippet (`hr-`) for inserting a horizontal rule. One
  of the defined syntax rules requires 80-character long series of
  dashes for horizontal rules in Markdown and this snippet eases using
  that construct.
- Defines recommended extensions for the project, including:

  - `reflow-markdown` for wrapping text more quickly.
  - `markdownlint` for checking Markdown syntax.
  - `github-local-actions` for running GHA locally before pushing.

michaeltlombardi commented:
### <span aria-hidden="true">✅</span> Deploy Preview for *astounding-sunflower-47e79a* ready!


|  Name | Link |
|:-:|------------------------|
|<span aria-hidden="true">🔨</span> Latest commit | 2d1cd6c20b2f3c4a2b56ec4b1b225dc6e1ffc0c8 |
|<span aria-hidden="true">🔍</span> Latest deploy log | https://app.netlify.com/projects/astounding-sunflower-47e79a/deploys/68c2fb4a4baf0100080988e8 |
|<span aria-hidden="true">😎</span> Deploy Preview | [https://deploy-preview-14--astounding-sunflower-47e79a.netlify.app](https://deploy-preview-14--astounding-sunflower-47e79a.netlify.app) |
|<span aria-hidden="true">📱</span> Preview on mobile | <details><summary> Toggle QR Code... </summary><br /><br />![QR Code](https://app.netlify.com/qr-code/eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1cmwiOiJodHRwczovL2RlcGxveS1wcmV2aWV3LTE0LS1hc3RvdW5kaW5nLXN1bmZsb3dlci00N2U3OWEubmV0bGlmeS5hcHAifQ.ywW_5AJ-Xyhi93AHfYwCtRpfi6T1j6CAz6xjIsAg44o)<br /><br />_Use your smartphone camera to open QR code link._</details> |
---
<!-- [astounding-sunflower-47e79a Preview](https://deploy-preview-14--astounding-sunflower-47e79a.netlify.app) -->
_To edit notification comments on pull requests, go to your [Netlify project configuration](https://app.netlify.com/projects/astounding-sunflower-47e79a/configuration/notifications#deploy-notifications)._

michaeltlombardi commented on content/blog/on-git-practices.md:
```diff
@@ -521,10 +521,10 @@ Until next time,
+- Report any goofs I've made, like typos or inaccuracies, [as a GitHub issue][ab].
 
-      And, if you do join the conversation or report a goof, know that I appreciate you.
+  And, if you do join the conversation or report a goof, know that I appreciate you.
```
These lines are indented so they correctly parse as being in the footnote - it's unfortunate and ugly, but changing them breaks the parsing.

Gijsreyn replied:
I was already a bit apprehensive about it. Now disabling it with inline comment (MD007)

Gijsreyn committed:
(MAINT) Fix rule violations in `on-git-practices.md`

Prior to this change, the initial blog post had some Markdown syntax
linting violations, which were uncovered by adding support for linting.

This change addresses those violations.
